reputation/updater.go

```diff
@@ -0,0 +1,302 @@
+		// check if user has votecaster nft pass
+		if !boosters.HasVotecasterNFTPass {
+			balance, err := u.airstack.Client.CheckIfHolder(VotecasterNFTPassAddress, VotecasterNFTPassChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting votecaster nft pass balance for %s: %w", addr, err))
+			} else {
+				boosters.HasVotecasterNFTPass = balance > 0
+			}
+		}
+		// check if user has votecaster launch nft
+		if !boosters.HasVotecasterLaunchNFT {
+			balance, err := u.airstack.Client.CheckIfHolder(VotecasterLaunchNFTAddress, VotecasterLaunchNFTChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting votecaster launch nft balance for %s: %w", addr, err))
+			} else {
+				boosters.HasVotecasterLaunchNFT = balance > 0
+			}
+		}
+		// check if user has KIWI
+		if !boosters.HasKIWI {
+			balance, err := u.census3.TokenHolder(KIWIAddress.Hex(), KIWIChainID, "", strAddr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting KIWI balance for %s: %w", addr, err))
+			} else {
+				boosters.HasKIWI = balance.Cmp(big.NewInt(0)) > 0
+			}
+		}
+		// check if user has DegenDAO NFT
+		if !boosters.HasDegenDAONFT {
+			balance, err := u.airstack.Client.CheckIfHolder(DegenDAONFTAddress, DegenDAONFTChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting DegenDAO NFT balance for %s: %w", addr, err))
+			} else {
+				boosters.HasDegenDAONFT = balance > 0
+			}
+		}
+		// check if user has 10k Degen
+		if !boosters.Has10kDegenAtLeast {
+			balance, err := u.airstack.Client.CheckIfHolder(DegenAddress, DegenChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting 10k Degen balance for %s: %w", addr, err))
+			} else {
+				boosters.Has10kDegenAtLeast = balance >= 10000
+			}
+		}
+		// check if user has TokyoDAO NFT
+		if !boosters.HasTokyoDAONFT {
+			balance, err := u.airstack.Client.CheckIfHolder(TokyoDAONFTAddress, TokyoDAONFTChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting TokyoDAO NFT balance for %s: %w", addr, err))
+			} else {
+				boosters.HasTokyoDAONFT = balance > 0
+			}
+		}
+		// check if user has Proxy and at least 5 Proxies
+		if !boosters.HasProxy {
+			balance, err := u.airstack.Client.CheckIfHolder(ProxyAddress, ProxyChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting Proxy balance for %s: %w", addr, err))
+			} else {
+				boosters.HasProxy = balance > 0
+				boosters.Has5ProxyAtLeast = balance >= 5
+			}
+		}
+		// check if user has NameDegen
+		if !boosters.HasNameDegen {
+			balance, err := u.airstack.Client.CheckIfHolder(NameDegenAddress, NameDegenChainShortName, addr)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("error getting NameDegen balance for %s: %w", addr, err))
+			} else {
+				boosters.HasNameDegen = balance > 0
+			}
+		}
+	}
+	if len(errs) > 0 {
```

Comment:
We should tolerate some errors here and do not update those user boosters (but do the rest).

So I'd receive a *Boosters struct as parameter and modify only those that have successfully been fetched. Those that fail, just keep the provided value (do not modify). Errors can just be log.Warn() printed